src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -471,40 +486,40 @@ public void loadStructure(@NotNull final String name, int rotateTimes, BlockPos
         }
     }
 
+    /**
+     * Iterates through all the required resources and stores them in the building.
+     */
     private void requestMaterials()
     {
-        //We need to deal with materials
-        if (!Configurations.builderInfiniteResources && job instanceof JobBuilder)
+        JobBuilder builderJob = (JobBuilder) job;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'builderJob' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)